### DISCUSSION TITLE
Won't work with NordVPN

### DISCUSSION THREAD

**Discussion by DoubleDutch87** [haugene/docker-transmission-openvpn#2102] [Q&A]
After setting up on my Synology NAS through Portainer, I am getting the following error: Options error: Unrecognized option or missing or extra parameter(s) in /etc/openvpn/nordvpn/hu.ovpn:1: html (2.4.7)

Whatever I try, it just won't work....
Is there a workaround?

**Comment by edgd1er** [comment]
No workaround, just read the logs to analyze what could be the problem: https://github.com/haugene/docker-transmission-openvpn/discussions/1993#discussioncomment-1786184

According to your logs conf is downloaded `Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/nl885.nordvpn.com.udp.ovpn` as the default configuration, which is a proper url. you can try to download  the file in a browser.

On the contrary, when trying to ̀download from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/hu.udp.ovpn`, you could realized that the url is incorrect ending to a 404 error. The error page is downloaded as a configuration file which it is not. openvpn stopped when loading the html page as a config file.

The question is why do the nordvpn api return a 404 page. Why is the download url malformed ? Could you please post your settings NORDVPN_ env values and the logs ?

the important line for you is best server: expected is <country iso code><num>.nordvpn.com
According to the logs, i'm afraid you don't have the last container version. Could you check that ?

I'm working on a new version for nordvpn script, here is more or less what you could expect when setting 
```
2021-12-10 22:34:07 Checking curl installation
2021-12-10 22:34:07 DNS: resolution ok
2021-12-10 22:34:07 PING: ok, configurations download site reachable
2021-12-10 22:34:07 Removing existing configs in /etc/openvpn/nordvpn
2021-12-10 22:34:07 Selecting the best server...
2021-12-10 22:34:07 Searching for country : hu (98)
2021-12-10 22:34:07 Searching for group: legacy_p2p
2021-12-10 22:34:07 Searching for technology: openvpn_tcp
2021-12-10 22:34:07 Best server : hu47.nordvpn.com
2021-12-10 22:34:07 Downloading config: hu47.nordvpn.com.ovpn
2021-12-10 22:34:07 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/hu47.nordvpn.com.tcp.ovpn```

**Comment by ilike2burnthing** [comment]
https://haugene.github.io/docker-transmission-openvpn/provider-specific/#nordvpn

**Comment by edgd1er** [comment]
Nordvpn expects 3 vars as written in haugene docs (cf link here upper), all can be defaulted with side effects. when no country is defined, home country is returned by the API. Nedelands seems to be your homeland, if you really need to have a vpn exiting frmo a foreign country you should pay attention to the logs.

In the last log,the config was downloaded as the link is correct. Did you try to download the config to check the link ?
At the moment, after reading the docs and set NORDVPN vars knowingly, you will have a DSM error:

Fri Dec 10 22:16:51 2021 ERROR: Cannot ioctl TUNSETIFF tun: Operation not permitted (errno=1)
Fri Dec 10 22:16:51 2021 Exiting due to fatal error

you then will have to read the topics specific to Synology DSM specific required settings. (pinne issue + doc )

**Comment by DoubleDutch87** [comment]


**Reply by edgd1er** [reply]
- Mullvad is another vpn provider.
- nordvpn config url is valid and was downloaded: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/nl721.nordvpn.com.tcp.ovpn
- the problem is now specific to synology DSM, which I don't have so I can't help you further. I suggest you read the doc, pinned issues, closed issues and open issues about that subject.
